src/Http/Controllers/Api/PtpController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Biigle\Modules\Ptp\Http\Controllers\Api;
+use Biigle\Http\Controllers\Api\Controller;
+use Biigle\Modules\Ptp\Jobs\PtpJob;
+use Biigle\Modules\Ptp\Jobs\UploadPtpExpectedAreaJob;
+use Biigle\Modules\Ptp\Jobs\UploadConvertedAnnotationsJob;
+use Biigle\Modules\Ptp\PtpExpectedArea;
+use Biigle\ImageAnnotation;
+use Biigle\Volume;
+use Biigle\Label;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Bus;
+
+/**
+ * Controller used for creating a PTP Job Chain
+ */
+class PtpController extends Controller
+{
+    /**
+     * Generate Point to Polygon Job Chain.
+     * This method generates, based on the request. The jobs generated are first for computing and uploading the expected areas of converted polygons.
+     * Then, it generates a job for executing the conversion and then upload the new annotations to the DB
+     *
+     * @param  $request
+     * @return
+     */
+    public function generatePtpJob(Request $request) {
+        $this->validate($request, ['label_id' => 'integer', 'volume_id' => 'integer']);
+        $volume = Volume::findOrFail($request->volume_id);
```

Comment:
You have to validate that the volume is an image volume and (maybe) that is does not have tiled images here. Video volumes are not supported.

@dlangenk Would the Python script work with a huge image mosaic?

Reply:
I haven't tried it and currently it wouldn't work on tiles. But if you would get the full mosaic and use the right image loading library I don't see anything that would speak against that.